From 8fa48db098e5f3bf3ee26a413b61c283195b8973 Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:13:09 -0400
Subject: [virt] kvm/ppc: book3s/pr - Default to big endian guest

Message-id: <1410765214-16377-54-git-send-email-dgibson@redhat.com>
Patchwork-id: 94774
O-Subject: [PATCH 53/78] KVM: PPC: Book3S PR: Default to big endian guest
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Signed-off-by: David Gibson <dgibson@redhat.com>

commit 94810ba4edc8fc49c68650306928245f6c0c99fa
Author: Alexander Graf <agraf@suse.de>
Date:   Thu Apr 24 13:04:01 2014 +0200

    KVM: PPC: Book3S PR: Default to big endian guest

    The default MSR when user space does not define anything should be identical
    on little and big endian hosts, so remove MSR_LE from it.

    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/book3s_pr.c b/arch/powerpc/kvm/book3s_pr.c
index 2fd5ff8..28ab0fe 100644
--- a/arch/powerpc/kvm/book3s_pr.c
+++ b/arch/powerpc/kvm/book3s_pr.c
@@ -1235,7 +1235,7 @@ static struct kvm_vcpu *kvmppc_core_vcpu_create_pr(struct kvm *kvm,
  kvmppc_set_pvr_pr(vcpu, vcpu->arch.pvr);
  vcpu->arch.slb_nr = 64;
 
- vcpu->arch.shadow_msr = MSR_USER64;
+ vcpu->arch.shadow_msr = MSR_USER64 & ~MSR_LE;
 
  err = kvmppc_mmu_init(vcpu);
  if (err < 0)
-- 
1.7.1